### DISCUSSION TITLE
I

### DISCUSSION THREAD

**Discussion by braindevices** [prisma/prisma#25410] [Q&A] [answered]
### Question

Is there a way to query Json field like SQL does with -> or ->> operators (used for accessing JSON paths in MySQL or PostgreSQL)?

I guess we can do raw query:

```
const result = await prisma.$queryRaw`
  SELECT *
  FROM \`e_store\`.\`products\`
  WHERE \`category_id\` = 1
  AND JSON_UNQUOTE(JSON_EXTRACT(\`attributes\`, '$.ports.usb')) > 0
  AND JSON_UNQUOTE(JSON_EXTRACT(\`attributes\`, '$.ports.hdmi')) > 0;
`;

```

But is there anyway to do this without raw query?

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @braindevices 👋 

For filtering on JSON fields, Prisma allows you to use the `path` option to select specific parts of the JSON value. This is supported for both PostgreSQL and MySQL, but the syntax differs slightly between the two databases. Do take a look at the [documentation](https://www.prisma.io/docs/orm/prisma-client/special-fields-and-types/working-with-json-fields#filter-on-a-json-field-advanced).
